js_modules/dagit/packages/core/src/assets/AssetMetadata.tsx

```diff
@@ -0,0 +1,53 @@
+import {gql} from '@apollo/client';
+import {Box, MetadataTable} from '@dagster-io/ui';
+import * as React from 'react';
+
+import {DAGSTER_TYPE_FRAGMENT} from '../dagstertype/DagsterType';
+import {DagsterTypeFragment} from '../dagstertype/types/DagsterTypeFragment';
+import {MetadataEntry, METADATA_ENTRY_FRAGMENT} from '../metadata/MetadataEntry';
+import {MetadataEntryFragment} from '../metadata/types/MetadataEntryFragment';
+import {__REPOSITORY_MEGA_JOB} from '../workspace/asset-graph/Utils';
+
+import {AssetNodeOpMetadataFragment} from './types/AssetNodeOpMetadataFragment';
+
+export const metadataForAssetNode = (
+  assetNode: AssetNodeOpMetadataFragment,
+): {assetType?: DagsterTypeFragment; assetMetadata: MetadataEntryFragment[]} => {
+  const assetType = assetNode.op?.outputDefinitions[0]?.type;
+  const assetMetadata = assetNode.op?.outputDefinitions[0]?.metadataEntries || [];
+  return {assetType, assetMetadata};
+};
```

Comment:
Hey @smackesey! I rebased this PR on top of your table metadata work and had to change up the queries a bit. I wanted to make this info load when you clicked on an asset in the graph to view it (rather than preloading it) because the global asset graph makes it a lot easier to have a ton of nodes.

I added a fragment for the `assetNode.op.outputDefinitions` data so that it didn't have to be repeated in two places, and made this helper that unwraps the response and returns the two inner types.

Reply:
Hey! I tested the type display, but I'm pulling your sean/bollinger to test the other cases again right now 👍  I didn't change much of the rendering but I did adjust padding in a few places when i moved the UI into the new two-tab design.